### DISCUSSION TITLE
Issue with tag-ignore with versioning

### DISCUSSION THREAD

**Discussion by adadkins** [community/community#25158] [Actions]
<p>We have a js monorepo and weve been using lerna to manage our versions. We tried to switch our CI/CD over to Github actions but running into an issue.</p>
<p>We have a work flow thats like: </p>
<pre><code>name: deploy
on:
    push:
        branches: [master]
        tags-ignore:
            - '*'
</code></pre>
<p>And one of the steps is to have lerna version and then put the build somewhere.</p>
<p>This works fine, however when lerna versions, it keeps track of the version in the package.json, so it will update the package.json with a newer value like “version”: “1.0.14”, then commit the package.json along with tags.</p>
<p>For some reason this is then causing the action to fire again, the tags-ignore is not working. I’ve tried all sorts of wildcards such as ‘**’ or ‘name*’ or ‘*.*’ all things that match the tags, and it still fires the action on the ‘update package.json/tag’ commit. I have a feeling its because this is a commit with a file change AND tagging, but I’m not sure. </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/adadkins">@adadkins</a> ,</p>
<p>Yes, this is by designed, commit the package.json along with tags will trigger the workflow.</p>
<p>Actually you can check the value of github.ref when the workflow triggered, if it shows like ‘refs/heads/Branchname’, it’s triggered by the file commit, if it shows like ‘refs/tags/v1.4.5’, it’s triggered by a tag.</p>
<pre><code>echo ${{ github.ref }}
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181089180-de6b22a7-626a-4789-9146-d7f36fef51bd.png" alt="refvalue.png" data-base62-sha1="kuKyjt8fgrOBjjRe4FAnnEMVVuP" width="271" height="98"></p>
<p>Thanks.</p>

**Comment by adadkins** [comment]
<p>Thank you, that was helpful. We were getting a tags ref and then a master ref as well.</p>
<p>If anyone else stumbles upon this post with a similar issue, you can add a logging step to your job and just echo the values to the console. </p>
<p>echo “1) message: ${{ github.event.head_commit.message }}”<br>
echo “2) event name: ${{ github.event_name }}”<br>
echo “3) ref: ${{ github.ref }}”</p>
<p>Which was really helful in determining what was causing the trigger and how we needed to fix it.</p>

**Comment by codeByunique** [comment]
Hey folks! 👋  
I'm Atabul, learning Java & Spring Boot. Just started exploring open source.  
Would love to hear how others found their first real contribution — any tips? 🙌
